packages/eas-cli/src/submissions/ios/AppSpecificPasswordSource.ts

```diff
@@ -0,0 +1,54 @@
+import chalk from 'chalk';
+import wordwrap from 'wordwrap';
+
+import log from '../../log';
+import { promptAsync } from '../../prompts';
+
+export enum AppSpecificPasswordSourceType {
+  userDefined,
+  prompt,
+}
+
+interface AppSpecificPasswordSourceBase {
+  sourceType: AppSpecificPasswordSourceType;
+}
+
+interface AppSpecificPasswordUserDefinedSource extends AppSpecificPasswordSourceBase {
+  sourceType: AppSpecificPasswordSourceType.userDefined;
+  appSpecificPassword: string;
+}
+
+interface AppSpecificPasswordPromptSource extends AppSpecificPasswordSourceBase {
+  sourceType: AppSpecificPasswordSourceType.prompt;
+}
+
+export type AppSpecificPasswordSource =
+  | AppSpecificPasswordUserDefinedSource
+  | AppSpecificPasswordPromptSource;
+
+export async function getAppSpecificPasswordAsync(source: AppSpecificPasswordSource) {
+  if (source.sourceType === AppSpecificPasswordSourceType.userDefined) {
+    return source.appSpecificPassword;
+  } else if (source.sourceType === AppSpecificPasswordSourceType.prompt) {
+    log.addNewLineIfNone();
+    const wrap = wordwrap(process.stdout.columns || 80);
+    log(
+      wrap(
+        chalk.bold(
+          `The password is only used to upload the build to Apple TestFlight and never stored on Expo servers`
```

Comment:
I think that in the future we will actually want to store the app-specific password on our servers. Treat this comment as a heads-up :)

Reply:
I'm not really sure that even now we can make that claim, technically when job is sent to submission service it's inside pub sub. Technically this is already storing it.

Reply:
ok so it's just safer to remove this claim for now.